### PR TITLE
[Modules] Updated Authorization/RoleAssignments to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Authorization/RoleAssignments to new dependency approach

| Pipeline |
| - |
| [Authorization: RoleAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.roleassignments.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

**Comment by ahmadabdalla** [comment]
Pending change to the implementation approach:
MicrosoftTeams-image

**Comment by MrMCake** [comment]
> Pending change to the implementation approach: MicrosoftTeams-image

As per our discussion I'd suggest to push this back as neither the global.module.tests, nor especially the Set-ModuleReadMe function will be able to properly handle this alternative approach. A larger refactoring may be required.

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'aramg'
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+module resourceGroupResources 'interm.dependencies.bicep' = {
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep:22]:
```suggestion
module resourceGroupResources 'interim.dependencies.bicep' = {
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,27 @@
+targetScope = 'subscription'
+
+@description('Required. The location to deploy to.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep:3]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+param location string
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep:4]:
```suggestion
param location string = deployment().location
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,27 @@
+targetScope = 'subscription'
+
+@description('Required. The location to deploy to.')
+param location string
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep:4]:
```suggestion
param location string = deployment().location
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg.min/dependencies.bicep [hunk 5]
```diff
@@ -0,0 +1,13 @@
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg.min/dependencies.bicep:4]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg.min/dependencies.bicep [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+param managedIdentityName string
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg.min/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+@description('Required. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string
+
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep:10]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested'
+  params: {
+    managedIdentityName: managedIdentityName
+  }
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg.min/interm.dependencies.bicep:26]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/dependencies.bicep [hunk 9]
```diff
@@ -0,0 +1,13 @@
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/dependencies.bicep:4]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+@description('Required. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string
+
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep:10]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested'
+  params: {
+    managedIdentityName: managedIdentityName
+  }
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/interm.dependencies.bicep:26]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep [hunk 12]
```diff
@@ -0,0 +1,13 @@
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep:1]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/rg/dependencies.bicep [hunk 13]
```diff
@@ -0,0 +1,13 @@
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/rg/dependencies.bicep:1]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/rg/dependencies.bicep [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/rg/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/sub.min/dependencies.bicep [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/sub.min/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/sub.min/dependencies.bicep [hunk 16]
```diff
@@ -0,0 +1,13 @@
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/sub.min/dependencies.bicep:1]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/sub/dependencies.bicep [hunk 17]
```diff
@@ -0,0 +1,13 @@
+@description('Required. The name of the managed identity to create.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/sub/dependencies.bicep:1]:
```suggestion
@description('Required. The name of the Managed Identity to create.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/sub/dependencies.bicep [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/sub/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.authorization.roleassignments-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'aramg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep:14]: aramgcom?
Also maybe rename the folder to be `mg.common`? This comment would apply to all scopes.

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]

File: modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep (hunk 19 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/roleAssignments/.test/mg/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created managed identity')
```
Comment (MrMCake) [inline modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity. ')
```

File: modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+@description('The principal ID of the created Managed Identity. ')
```
Comment (MrMCake) [inline modules/Microsoft.Authorization/roleAssignments/.test/rg.min/dependencies.bicep:12]:
```suggestion
@description('The principal ID of the created Managed Identity.')
```

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**